pkgs/development/python-modules/extractcode/libarchive.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, libarchive
+, libb2
+, bzip2
+, expat
+, lz4
+, lzma
+, zlib
+, zstd
+}:
+buildPythonPackage rec {
+  pname = "extractcode-libarchive";
+  version = "21.4.4";
+
+  src = fetchFromGitHub {
+    owner = "nexB";
+    repo = "scancode-plugins";
+    rev = "v${version}";
+    sha256 = "xnUGDMS34iMVMGo/nZwRarGzzbj3X4Rt+YHvvKpmy6A=";
+  };
+
+  postPatch = ''
+    cd builtins/extractcode_libarchive-linux
+  '';
+
+  preBuild = ''
+    pushd src/extractcode_libarchive/lib
+
+    cp ${lib.getLib libarchive}/lib/libarchive.so libarchive.so
+    cp ${lib.getLib libb2}/lib/libb2.so libb2-la3511.so.1
+    cp ${lib.getLib bzip2}/lib/libbz2.so libbz2-la3511.so.1.0
+    cp ${lib.getLib expat}/lib/libexpat.so libexpat-la3511.so.1
+    cp ${lib.getLib lz4}/lib/liblz4.so liblz4-la3511.so.1
+    cp ${lib.getLib lzma}/lib/liblzma.so liblzma-la3511.so.5
+    cp ${lib.getLib zlib}/lib/libz.so libz-la3511.so.1
+    cp ${lib.getLib zstd}/lib/libzstd.so libzstd-la3511.so.1
```

Comment:
This will only work on Linux. Adjust `meta.platforms`.

However, copying those libraries doesn't seem like a good solution. We should link to them.

Reply:
Ah, I thought I changed this to `ln -s` when I did the same to `-7z`, whoops. Thanks.

Reply:
Please check that using `ln -s` actually reduces the closure size.

Reply:
@cole-h Did you?

Reply:
There's no difference.

```
$ nix path-info -Sh ./result
/nix/store/2sm76cdmm7nxnn60gw9sdm8anp4lij9i-python3.8-extractcode-libarchive-21.4.4      112.8M
$ nix path-info -Sh ./result
/nix/store/l81z1qhrs4nz5y9ga484qlddh34q6z4w-python3.8-extractcode-libarchive-21.4.4      112.8M

Reply:
Also note that they get installed as the real file (e.g. not a symlink in `result/lib/python3.8/site-packages/extractcode_libarchive/lib`). But rather than create more commit churn, I'd like to leave as-is (symlinking before build).